### PR TITLE
Feature/terms check mark

### PR DESCRIPTION
**Pull request by priyesh777**
## Changes:
- Add checkbox to accept terms and conditions in register section

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/views/Register/RegisterForm/index.tsx [hunk 1]
```diff
@@ -85,6 +91,12 @@ function RegisterModal(props: Props) {
 
     const error = getErrorObject(riskyError);
 
+    const [
+        showTermsModal, ,
```
Comment (AdityaKhatri) [inline app/views/Register/RegisterForm/index.tsx]:
```suggestion
        isTermsModalShown, ,
```

File: app/views/Register/RegisterForm/index.tsx [hunk 2]
```diff
@@ -116,99 +128,155 @@ function RegisterModal(props: Props) {
         },
     );
 
+    const handleTermsCheck = useCallback(() => {
+        setAcceptTerms(!acceptTerms);
+    }, [acceptTerms]);
```
Comment (AdityaKhatri) [inline app/views/Register/RegisterForm/index.tsx]: not needed

File: app/views/Register/RegisterForm/index.tsx [hunk 3]
```diff
@@ -116,99 +128,155 @@ function RegisterModal(props: Props) {
...
+                            >
+                                Accept
+                            </Button>
+                            <Button
+                                name={undefined}
+                                onClick={hideTermsModal}
+                                variant="secondary"
+                            >
+                                Close
+                            </Button>
```
Comment (AdityaKhatri) [inline app/views/Register/RegisterForm/index.tsx]: change order

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/Register/RegisterForm/index.tsx [hunk 4]
```diff
@@ -73,6 +78,7 @@ function RegisterModal(props: Props) {
 
     const elementRef = useRef<Captcha>(null);
     const [success, setSuccess] = useState(false);
+    const [acceptTerms, setAcceptTerms] = useState(false);
```
Comment (tnagorra) [inline app/views/Register/RegisterForm/index.tsx]:
```suggestion
    const [termsAccepted, setTermsAffected] = useState(false);
```

File: app/views/Register/RegisterForm/index.tsx [hunk 5]
```diff
@@ -85,6 +91,12 @@ function RegisterModal(props: Props) {
 
     const error = getErrorObject(riskyError);
 
+    const [
+        isTermsModalShown, ,
+        isTermsModalHidden, ,
```
Comment (tnagorra) [inline app/views/Register/RegisterForm/index.tsx]:
```suggestion
        setTermsModalHidden, ,
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: app/views/Register/RegisterForm/index.tsx (hunk 4 above)
Comment (priyesh777) [reply app/views/Register/RegisterForm/index.tsx]: I hope that means "setTermsAccepted"

**Pull request merged**